src/vm/vm_memory/memory_segments.rs

```diff
@@ -96,32 +98,73 @@ impl MemorySegmentManager {
         Ok(relocation_table)
     }
 
-    pub fn gen_arg_vec_bigint(
-        &self,
-        arg: &[BigInt],
+    pub fn gen_arg(
+        &mut self,
+        arg: &dyn Any,
```

Comment:
Wouldn't that change the functionality?

For example, passing `[5]` to starkware's `gen_arg()` would generate a new segment with that value while our implementation would apply modulo and return a `BigInt` instead of a `Relocatable` pointing to the new segment.

Starkware implementation:
```py
def gen_arg(self, arg, apply_modulo_to_args=True) -> MaybeRelocatable:
    """
    Converts args to Cairo-friendly ones.
    If an argument is Iterable it is replaced by a pointer to a new segment containing the items
    in the Iterable arg (recursively).
    If apply_modulo_to_args=True, all the integers are taken modulo the program's prime.
    """
    if isinstance(arg, Iterable):
        base = self.add()
        self.write_arg(base, arg)
        return base
    if apply_modulo_to_args and isinstance(arg, int):
        return arg % self.prime
    return arg
```